src/shared/sleep-config.c

```diff
@@ -102,6 +102,107 @@ int parse_sleep_config(SleepConfig **ret_sleep_config) {
         return 0;
 }
 
+int battery_is_low(void) {
+        int r;
+
+        r = read_battery_capacity_percentage();
+        if (r < 0)
+               return r;
+
+        if (r > 0 && r <= 5)
+               return 1;/*If battery current capacity percentage is equal to or less than 5%. We have not used battery capacity_level since value is set to full or Normal in case acpi is not working properly.*/
+
+        return 0;/*Not low if greater than 5%. The error will be handled in read_battery_capacity_percentage which returns value from 0-100 only*/
+}
+
+int read_battery_capacity_percentage(void) {
+        _cleanup_free_ char *bat_cap = NULL;
+        int battery_capacity, r;
+
+        r = read_one_line_file("/sys/class/power_supply/BAT0/capacity", &bat_cap);
+        if (r == -ENOENT)
```

Comment:
this shouldn't be handled differently here, ie just log_debug_errno as usual, it's the caller that needs to pay attention. The debug message is still useful.

Reply:
Yes makes sense. It is handled when it is called twice in sleep.c. Adding debug err log instead of returning err code.